operator/v1/types.go

```diff
@@ -382,6 +382,14 @@ type KlusterletSpec struct {
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	ClusterName string `json:"clusterName,omitempty"`
 
+	// InitClusterLabels is a set of labels that are applied to the managed cluster created during the registration process.
+	// +optional
+	InitClusterLabels map[string]string `json:"initClusterLabels,omitempty"`
```

Comment:
can use change this field, what if user update it? And should we ensure user cannot set arbitrary key on that?

Reply:
The fields are assumed only to be used in the "init" in here: https://github.com/open-cluster-management-io/ocm/blob/41f430b0776a4a7dbb1006a2125838e2994c4b17/pkg/registration/spoke/registration/creating_controller.go#L65

After the bootstrap secret expired, we will never enter this block of code again, so "update" operation will do nothing.

--

I'm not sure what restrictions we need in this case. What key we expect user not use?

Reply:
for instance clusterset label key. init is not quite clear, this only happens when the agent is trying to create the cluster. If user delete the cluster on the hub, and agent will recreate the cluster use this label. So we cannot say it is only used during init.

Reply:
and I think it should be put under registration configuration field

Reply:
Got it, let me think it through again.

Reply:
Fixed, @qiujian16  PTAL